src/theory/booleans/circuit_propagator.cpp

```diff
@@ -716,7 +716,9 @@ TrustNode CircuitPropagator::propagate()
                     && (current[0].isVar() && current[1].isVar()));
 
     // If an atom, add to the list for simplification
-    if (atom)
+    if (atom
```

Comment:
Can we just refactor the definition of atom to do this?  I actually think the 3rd case of atom above is a stronger version of this?

Reply:
The PR doesn't change `atom` because it is used in a different context below. The idea is that equalities between Boolean variables and terms are treated as theory literals for the purpose of learning them but otherwise as regular Boolean terms.